### PR TITLE
delara/eng-1037-fe-filter-contacts-based-on-statuses

### PR DESCRIPTION
**Pull request by Mittensdlara**
# Description

In `contacts/index.ts` there is a table and all contacts have a statuses, so there we need a `select box` to enable us filter the contacts base on their `status.src/pages/cotacts/index.ts`

Noticed that first of all you need to make a separate component for `select box` .

Feat # (1037)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

By using `cosole.log` and checking out the UI of the App.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: frontend/src/pages/contacts/ContactStatusSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import { SelectFormControl } from '../../components/molecules/formControls/selectFormControl';
+import { uri } from '../../utils';
+
+interface Props {
+	scoreDigit: any;
+}
```
Comment (hrahimi270) [inline frontend/src/pages/contacts/ContactStatusSelect.tsx]: The `scoreDigit` type should be relative to the function. Don't use the `any` type for functions.
You could use a type like this for it:
```js
interface Props {
    scoreDigit(results: any): void;
}
```
You could also make sure about the `results` type too and use the relative type for it if there is any.

File: frontend/src/pages/contacts/ContactStatusSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+			const response = await axios.get(url, {
+				headers: {
+					Authorization: `Bearer ${localStorage.getItem('access_token')}`,
+				},
+			});
+			scoreDigit(response.data);
+			console.log(response.data.contacts);
+		} catch (error) {
+			console.error(error);
+		}
```
Comment (hrahimi270) [inline frontend/src/pages/contacts/ContactStatusSelect.tsx]: We don't use console logs for production. You are allowed to use them while you are developing. But remove them before pushing your code.

Comment (Mittensdlara) [reply frontend/src/pages/contacts/ContactStatusSelect.tsx]: Thanks, I'll fix it.

File: frontend/src/pages/contacts/ContactStatusSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+			const response = await axios.get(url, {
+				headers: {
+					Authorization: `Bearer ${localStorage.getItem('access_token')}`,
+				},
+			});
+			scoreDigit(response.data);
+		} catch (err: any) {
+			setError(err?.response?.data?.message);
+			if (err.message) setError(err.message);
+		}
```
Comment (hrahimi270) [inline frontend/src/pages/contacts/ContactStatusSelect.tsx]: You're setting the `setError` two times. It's better to put the first `setError` inside the `else` of the if.

**Review by hrahimi270 — APPROVED** [review]

**Pull request merged**